### PR TITLE
RestAPI start download endpoint

### PR DESCRIPTION
**Pull request by brussee**
This problematic wrapper function is now unused:  (a flag is used instead by @devos50)
```
    @waitForHandleAndSynchronize(True)
    def set_share_mode(self, share_mode):
        self.handle.set_share_mode(share_mode)
```

### CODE REVIEW

**Comment by brussee** [comment]
@lfdversluis Would you review this please?

File: Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py [hunk 1]
```diff
@@ -304,7 +302,11 @@ def network_create_engine_wrapper(self, pstate, initialdlstatus=None, checkpoint
 
             self.handle = self.ltmgr.add_torrent(self, atp)
 
-            if self.handle:
+            # DEBUG:
```
Comment (devos50) [inline Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py]: Don't forget to remove this debug statement if it's stable ;)

File: Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py [hunk 2]
```diff
@@ -1173,10 +1181,6 @@ def dlconfig_changed_callback(self, section, name, new_value, old_value):
     def get_share_mode(self):
         return self.handle.status().share_mode
 
-    @waitForHandleAndSynchronize(True)
-    def set_share_mode(self, share_mode):
```
Comment (devos50) [inline Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py]: Note that @ardhipoetra has to change several calls in his credit mining code if we do this.

**Comment by devos50** [comment]
@brussee good work 👍

File: Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py (hunk 2 above)
Comment (brussee) [inline Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py]: Yes, I thought about putting it back. The use of a flag on_add is working as far as the assert can guarantee.

But I am wondering if this method is stable enough to use after the torrent has been added.
@ardhipoetra Is the one to find that one out.

File: Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py (hunk 1 above)
Comment (brussee) [inline Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py]: Yes, I would like to invite @ardhipoetra to use this in his PR #2399.

**Comment by brussee** [comment]
@devos50 Thanks! You too!

File: Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py (hunk 2 above)
Comment (ardhipoetra) [inline Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py]: AFAIK, this method is stable enough even without decorator as long as the handle exist (and ready). In the early time I implemented this I have no issue with this.

File: Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py (hunk 1 above)
Comment (ardhipoetra) [inline Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py]: noted! :)

File: Tribler/Test/Core/Libtorrent/test_libtorrent_download_impl.py [hunk 3]
```diff
@@ -185,20 +200,6 @@ def test_get_share_mode(self):
         self.libtorrent_download_impl.handle.status().share_mode = True
         self.assertTrue(self.libtorrent_download_impl.get_share_mode())
 
-    def test_set_share_mode(self):
```
Comment (ardhipoetra) [inline Tribler/Test/Core/Libtorrent/test_libtorrent_download_impl.py]: are you deleting this test as well as the wrapper function?

File: Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py (hunk 2 above)
Comment (brussee) [inline Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py]: Alright I will put it back, use at own risk!

File: Tribler/Test/Core/Libtorrent/test_libtorrent_download_impl.py (hunk 3 above)
Comment (brussee) [inline Tribler/Test/Core/Libtorrent/test_libtorrent_download_impl.py]: I will put it back, but use at own risk!

**Comment by brussee** [comment]
@whirm Ready.

**Pull request merged**